.github/actions/deploy/deploy.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+set -e -u
+
+# this was based off of the deploy script in the mbta/actions/deploy-ecs
+# action.
+#
+# Register a new ECS Task Definition using the new Docker Image
+# Update the ECS Service with this new Task Definition
+#
+# Required Arguments
+# ECS_CLUSTER: ecs cluster for service being updated
+# ECS_SERVICE: ecs service containing task being updated
+# ECS_TASK_DEF: task definition to update revision for
+# DOCKER_TAG: tag for docker image to use in new task definition
+
+# set default region so we don't have to specify --region everywhere
+export AWS_DEFAULT_REGION="us-east-1"
+
+# get the contents of the template task definition from ECS 
+echo "Retrieving ${ECS_TASK_DEF}-template task definition..."
+taskdefinition="$(aws ecs describe-task-definition --task-definition "${ECS_TASK_DEF}-template")"
+
+# use retrieved task definition as basis for new revision, but replace image
+echo "Updating container image to ${DOCKER_TAG}."
+newcontainers="$(echo "${taskdefinition}" | \
+  jq '.taskDefinition.containerDefinitions' | \
+  jq --arg tag "${DOCKER_TAG}" 'map(.image="\($tag)")')"
+
+# check to make sure the secrets are included in the new container definition
+if (echo "${newcontainers}" | jq '.[0] | .secrets' | grep '^null$'); then
+  echo "Error: The container definition is missing its 'secrets' block. Deploy cannot proceed."
+  exit 1
+fi
+
+# this task uses fargate, so publish the new task definition
+echo "Publishing new Task Definition."
+aws ecs register-task-definition \
+--family "${ECS_TASK_DEF}" \
+--task-role-arn "$(echo "${taskdefinition}" | jq -r '.taskDefinition.taskRoleArn')" \
+--execution-role-arn "$(echo "${taskdefinition}" | jq -r '.taskDefinition.executionRoleArn')" \
+--network-mode "$(echo "${taskdefinition}" | jq -r '.taskDefinition.networkMode')" \
+--container-definitions "${newcontainers}" \
+--volumes "$(echo "${taskdefinition}" | jq '.taskDefinition.volumes')" \
+--placement-constraints "$(echo "${taskdefinition}" | jq '.taskDefinition.placementConstraints')" \
+--requires-compatibilities "$(echo "${taskdefinition}" | jq -r '.taskDefinition.requiresCompatibilities')" \
+--cpu "$(echo "${taskdefinition}" | jq -r '.taskDefinition.cpu')" \
+--memory "$(echo "${taskdefinition}" | jq -r '.taskDefinition.memory')"
+
+# get the new revision of this task and redeploy the cluster with it
+newrevision="$(aws ecs describe-task-definition --task-definition "${ECS_TASK_DEF}" | \
+  jq -r '.taskDefinition.revision')"
+
+echo "Updating Service ${ECS_SERVICE} to use Task Definition ${newrevision}..."
+aws ecs update-service --cluster="${ECS_CLUSTER}" --service="${ECS_SERVICE}" --task-definition "${ECS_TASK_DEF}:${newrevision}"
```

Comment:
removed checks that the running task count is up to 1 and check that older deployments have been been stopped. that shouldn't be a concern for us as the task count is always going to be 0 around deployments.

i did think that we could check that the deployment is complete by looking at the `ecs describe-service` command, which results in something like

```
❯ aws ecs describe-services --cluster lamp --services lamp-ingestion-staging | jq -r '.services'
[
  {
    "serviceArn": "arn:aws:ecs:us-east-1:434035161053:service/lamp/lamp-ingestion-staging",
    "serviceName": "lamp-ingestion-staging",
    "clusterArn": "arn:aws:ecs:us-east-1:434035161053:cluster/lamp",
    "loadBalancers": [],
    "serviceRegistries": [],
    "status": "ACTIVE",
    "desiredCount": 1,
    "runningCount": 1,
    "pendingCount": 0,
    "launchType": "FARGATE",
    "platformVersion": "LATEST",
    "platformFamily": "Linux",
    "taskDefinition": "arn:aws:ecs:us-east-1:434035161053:task-definition/lamp-ingestion-staging:91",
    "deploymentConfiguration": {
      "deploymentCircuitBreaker": {
        "enable": false,
        "rollback": false
      },
      "maximumPercent": 100,
      "minimumHealthyPercent": 0
    },
    "deployments": [
      {
        "id": "ecs-svc/8666230551424498921",
        "status": "PRIMARY",
        "taskDefinition": "arn:aws:ecs:us-east-1:434035161053:task-definition/lamp-ingestion-staging:91",
        "desiredCount": 1,
        "pendingCount": 0,
        "runningCount": 1,
        "failedTasks": 0,
        "createdAt": "2023-11-01T15:39:42.776000-04:00",
        "updatedAt": "2023-11-01T15:43:52.868000-04:00",
        "launchType": "FARGATE",
        "platformVersion": "1.4.0",
        "platformFamily": "Linux",
        "networkConfiguration": {
          "awsvpcConfiguration": {
            "subnets": [
              "subnet-0ecf10ef54256e988",
              "subnet-09525d49bf33e8caa",
              "subnet-0b00903afc1dc8019"
            ],
            "securityGroups": [
              "sg-0206b82c4667d6560"
            ],
            "assignPublicIp": "DISABLED"
          }
        },
        "rolloutState": "COMPLETED",
        "rolloutStateReason": "ECS deployment ecs-svc/8666230551424498921 completed."
      }
    ],
```
and we can compare the desk definition with the task definition of the only deployment and that the deployment is at a "COMPLETED" rollout.